net/libproxy/Portfile

```diff
@@ -18,94 +18,53 @@ long_description    Libproxy exists to answer the question: Given \
-#
-# webkit pacrunner circular dependency problem
-# webkit-gtk -> libsoup -> libproxy -> webkit-gtk (#26261)
-#
-# pac files support not available on Tiger (#24127)
-#
-# TODO: add variants that safely enable these features
-
 use_parallel_build no
 
-# use default GNOME build which targets GNOME3 (gsettings) instead of GNOME2 (gconf)
-# perl bindings disabled due to unpredictable configuration
-# kde binding (KDE4/KF5) tests at runtime for kreadconfig5 kreadconfig in that order
-
 configure.args-append \
-                    -DMP_MACOSX=NO \
-                    -DWITH_WEBKIT=NO \
-                    -DWITH_WEBKIT3=NO \
-                    -DWITH_VALA=YES \
-                    -DWITH_PERL=NO \
-                    -DWITH_PYTHON2=NO \
-                    -DWITH_PYTHON3=YES \
-                    -DWITH_KDE=YES
+                    -Ddocs=false \
+                    -Dtests=false \
+                    -Dconfig-env=false \
+                    -Dconfig-gnome=true \
+                    -Dconfig-kde=true \
+                    -Dconfig-windows=false \
+                    -Dconfig-sysconfig=false \
+                    -Dconfig-osx=false \
+                    -Dconfig-xdp=false \
+                    -Dpacrunner-duktape=true \
+                    -Dvapi=false \
+                    -Dcurl=false \
+                    -Dintrospection=true \
+                    -Drelease=true
 
 # Fix an issue with clang-900 (Xcode 9) using the Sierra SDK, <rdar://problem/31263056>
 configure.cppflags-append -D_DARWIN_C_SOURCE=1
 
-# Set standard via cxx flags, as cmake.set_cxx_standard is not working properly
-configure.cxxflags-append -std=c++11
-
-variant python39 conflicts python310 python311 python312 description {Build Bindings for Python 3.9} {
-    set python_prefix ${frameworks_dir}/Python.framework/Versions/3.9
-    depends_lib-append port:python39
-    configure.args-append \
-                    -DPYTHON3_EXECUTABLE=${prefix}/bin/python3.9 \
-                    -DPYTHON3_SITEPKG_DIR=${python_prefix}/lib/python3.9/site-packages
-}
-
-variant python310 conflicts python39 python311 python312 description {Build Bindings for Python 3.10} {
-    set python_prefix ${frameworks_dir}/Python.framework/Versions/3.10
-    depends_lib-append port:python310
+variant duktape description {Build with support for PAC Runner Duktape} {
+    depends_lib-append port:duktape
     configure.args-append \
-                    -DPYTHON3_EXECUTABLE=${prefix}/bin/python3.10 \
-                    -DPYTHON3_SITEPKG_DIR=${python_prefix}/lib/python3.10/site-packages
+                    -Dpacrunner-duktape=true
 }
 
-variant python311 conflicts python39 python310 python312 description {Build Bindings for Python 3.11} {
-    set python_prefix ${frameworks_dir}/Python.framework/Versions/3.11
-    depends_lib-append port:python311
+variant vapi description {Build with support for vapi} {
+    depends_lib-append port:vala
     configure.args-append \
```

Comment:
All these should be `configure.args-replace`, because default configure args above have `-Dvapi=false` (and others), and variant needs to change those, not  add a contradicting argument.

And path-style `vala`.

So, for example:
```
depends_build-append path:bin/vala:vala
configure.args-replace -Dvapi=false -Dvapi=true
```
(I think Vala is a build dependency, please check that.)

Reply:
All fixed

Reply:
I think you're right re: build dependency. It appears that enabling the option only results in a call to generate some auxiliary VAPI file

- https://github.com/libproxy/libproxy/blob/main/src/libproxy/meson.build#L82
- https://mesonbuild.com/Gnome-module.html#gnomegenerate_vapi